src/linters/tsc.js

```diff
@@ -0,0 +1,94 @@
+const { run } = require("../utils/action");
+const commandExists = require("../utils/command-exists");
+const { initLintResult } = require("../utils/lint-result");
+const { getNpmBinCommand } = require("../utils/npm/get-npm-bin-command");
+const { removeTrailingPeriod } = require("../utils/string");
+
+/** @typedef {import('../utils/lint-result').LintResult} LintResult */
+
+/**
+ * https://www.typescriptlang.org/docs/handbook/compiler-options.html
+ */
+class TSC {
+	static get name() {
+		return "TypeScript";
+	}
+
+	/**
+	 * Verifies that all required programs are installed. Throws an error if programs are missing
+	 * @param {string} dir - Directory to run the linting program in
+	 * @param {string} prefix - Prefix to the lint command
+	 */
+	static async verifySetup(dir, prefix = "") {
+		// Verify that NPM is installed (required to execute ESLint)
+		if (!(await commandExists("npm"))) {
+			throw new Error("NPM is not installed");
+		}
+
+		// Verify that ESLint is installed
+		const commandPrefix = prefix || getNpmBinCommand(dir);
+		try {
+			run(`${commandPrefix} tsc -v`, { dir });
+		} catch (err) {
+			throw new Error(`${this.name} is not installed`);
+		}
+	}
+
+	/**
+	 * Runs the linting program and returns the command output
+	 * @param {string} dir - Directory to run the linter in
+	 * @param {string[]} extensions - File extensions which should be linted
+	 * @param {string} args - Additional arguments to pass to the linter
+	 * @param {boolean} fix - Whether the linter should attempt to fix code style issues automatically
+	 * @param {string} prefix - Prefix to the lint command
+	 * @returns {{status: number, stdout: string, stderr: string}} - Output of the lint command
+	 */
+	static lint(dir, extensions, args = "", fix = false, prefix = "") {
+		// TSC does not support auto-fixing
```

Comment:
```suggestion
		if (fix) {
			core.warning(`${this.name} does not support auto-fixing`);
		}
```

This should issue a warning in this case.